clightning/config.go

```diff
@@ -271,11 +271,16 @@ func BitcoinFallbackFromClnConfig(client *ClightningClient) Processor {
 					}
 					if v, ok := plugin.Options["bitcoin-rpcport"]; ok {
 						if v != nil {
-							port, err := strconv.Atoi(v.(string))
-							if err != nil {
-								return nil, err
+							// detect if type is string (CLN < v23.08)
+							if reflect.TypeOf(v).String() == "string" {
+								port, err := strconv.Atoi(v.(string))
+								if err != nil {
+									return nil, err
+								c.Bitcoin.RpcPort = uint(port)
+								}
+							} else {
+								c.Bitcoin.RpcPort = uint(v.(float64))
```

Comment:
Nice that you also thought about backwards compatibility!
Instead of reflection (which is more powerful but also requires more memory and is slower) we could take the direct and idiomatic way, using type assertions. This way we could use a "type switch" that could also tell us about cases we missed or that we have not covered on future changes of external apis.
```suggestion
							// detect if type is string (CLN < v23.08)
							switch p := v.(type) {
							case string:
							        port, err := strconv.Atoi(p)
								if err != nil {
									return nil, err
							        }
							        c.Bitcoin.RpcPort = uint(port)
							case float64:
							        c.Bitcoin.RpcPort = uint(p)
							default:
							        return nil, fmt.Errorf("Bitcoind rpcport type %T not handled", v)
							}
```